server/src/main/java/org/opensearch/index/shard/IndexShard.java

```diff
@@ -4314,7 +4314,7 @@ public void close() throws IOException {
 
         // When the new engine is created, translogs are synced from remote store onto local. Since remote store is the source
         // of truth for translog, we play all translogs that exists locally. Otherwise, the recoverUpto happens upto global checkpoint.
-        long recoverUpto = this.isRemoteTranslogEnabled() ? Long.MAX_VALUE : globalCheckpoint;
+        long recoverUpto = this.isRemoteTranslogEnabled() || indexSettings().isSegRepEnabled() ? Long.MAX_VALUE : globalCheckpoint;
```

Comment:
nit: Update comment providing details around why change is needed for segment replication ?